### PR TITLE
tried to implement login and register pages

### PR DESCRIPTION
**Pull request by vitakh**
I tried to implement login and register pages by adding error handlers. However, even if I disabled some actions by returning 403 status, I still could not figure out how to display it for users for some of the errors. One of them is when dealing with already existent username. 
Another thing I am missing is the implementation of the admin role because I am currently struggling with connecting checkboxes with roles.
TODO: more edge cases for login and registration, handling roles

### CODE REVIEW

**Comment by raymondeah** [comment]
The .idea and node_modules directories should be left out of the repo. Could you add these to the gitignore and push another commit?

File: src/users/register.js [hunk 1]
```diff
@@ -6,10 +6,30 @@ import {Navigate} from "react-router";
...
+        if (password.length < 2 || password.length > 20) {
+            setError('Password should be between 2 and 20 characters')
+            return
+        }
+        if (password !== validatePassword) {
+            setError('Passwords must match')
+            return
+        }
+        setError(null)
         dispatch(registerThunk({username, password}))
```
Comment (raymondeah) [inline src/users/register.js]: If the checkbox is working correctly then you can modify the registerThunk and any related functions to pass in the role as well

```
admin ? dispatch(registerThunk({username, password, role: 'ADMIN'}) : dispatch(registerThunk({username, password, role: 'USER'})
```

File: src/users/register.js [hunk 2]
```diff
@@ -6,10 +6,30 @@ import {Navigate} from "react-router";
 
 const Register = () => {
     const {currentUser} = useSelector((state) => state.users)
-    const [username, setUsername] = useState('alice')
-    const [password, setPassword] = useState('alice1234')
+    const [username, setUsername] = useState("")
+    const [password, setPassword] = useState("")
+    const [validatePassword, setValidatePassword] = useState("")
+    const [role, setRole] = useState(false)
```
Comment (raymondeah) [inline src/users/register.js]: I would change the naming to admin, setAdmin

**Comment by raymondeah** [comment]
On the first point, Could you elaborate more on what errors you are trying to deal with and what issues you are running into?

**Review by vitakh — COMMENTED** [review]
I implemented the admin role option, so now whenever the checkbox is checked, the user must be able to register as admin. 
More elaboration on the possible errors: How can we communicate the error 403 message that a user already exists, for example?

**Comment by raymondeah** [comment]
We could have a notification pop up whenever a 403 comes back, there's a lot of libraries that can handle it, bootstrap has [these](https://getbootstrap.com/docs/4.3/components/toasts/)

**Comment by likuny0415** [comment]
All the conflicts in this branch have been solved manually.

**Pull request closed without merging**